torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -269,6 +270,42 @@ def from_float_static(
             dtype=input_float.dtype,
         )
 
+    @classmethod
+    def from_float_float8(
```

Comment:
@jerryzh168  can you help with a design on how to have `from_float`, `from_float_static`, etc extend to this use case?  Ideally we shouldn't special case a set of dtypes (float8) to have their own function.

Reply:
A combined function is better, will be refactoring it after testing float8

Reply:
yeah sure, I think we could have the two following final state:
* have separate `from_float_fpx` and `from_from_intx` since they have a bit different arg list
*  if we manage to generalize the arg list enough so it is reasonable merge the two then we can merge as well, I will discuss with Apurva and Driss about the args but at the first glance maybe preserve_zero is always going to be true and zero_point_domain may not apply here

Reply:
makes sense.

one thought, the `from_float` name will become more confusing if both the source and the target can also be various floating point bitwidths.  To clarify this in `torchao.float8`, I went with the `high_precision|hp` and `low_precision|lp` naming scheme

Reply:
This is also why I like the idea of extending `to` and having our own factory functions that we can pass dtype enums.

Reply:
@vkuzo yeah makes sense, we can rename `from_float` to `from_high_precision` as well. as @cpuhrsch mentioned, this is not user facing API, we'll have `to` factory functions for various dtypes as the user facing API: https://github.com/pytorch/ao/blob/5c0e060a1a204d51ce6bc09901bc6f0574d2e38c/torchao/dtypes/affine_quantized_tensor.py#L990-L991

Reply:
@jainapurva as discussed from the meeting, let's merge this into `from_float` and add some guards on arguments